### PR TITLE
lecture 9 - drag and drop

### PR DESCRIPTION
**Pull request by IzabellaSantos**
Drag and drop feature was added

### CODE REVIEW

**Review by brunogvn — APPROVED** [review]

File: EmojiArt/EmojiArtDocumentView.swift [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+//  EmojiArt
+//
+//  Created by Izabella Julia dos Santos on 20/06/22.
+//
+
+import SwiftUI
+
+struct EmojiArtDocumentView: View {
+    @ObservedObject var document: EmojiArtDocument
+    let defaultEmojiFontSize: CGFloat = 40
```
Comment (RenatoMAssis) [inline EmojiArt/EmojiArtDocumentView.swift:12]: If it's not being used outside the struct we could set to private

File: EmojiArt/EmojiArtDocumentView.swift [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    private func fontSize(for emoji: EmojiArtModel.Emoji) -> CGFloat {
+        CGFloat(emoji.size)
+    }
+    
+    var palette: some View {
+        ScrollingEmojisView(emojis: testEmojis)
+            .font(.system(size: defaultEmojiFontSize))
+    }
+    
+    let testEmojis = "🐮🐱🐭🐹🐰🦊🐻🐯🦁🐷🐶🐐🦌🐕🐩🦮🐕‍🦺🐈🐈‍⬛🐓🦃🦤🦚🦜🦢🦩🕊🐇🦝🦨🦡🦫🦦🦥🐁🐀🐿🦔"
```
Comment (RenatoMAssis) [inline EmojiArt/EmojiArtDocumentView.swift:75]: If it's not being used outside the struct we could set to private

File: EmojiArt/EmojiArtDocumentView.swift [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+}
+
+struct ScrollingEmojisView: View {
+    let emojis: String
```
Comment (RenatoMAssis) [inline EmojiArt/EmojiArtDocumentView.swift:79]: If it's not being used outside the struct we could set to private

File: EmojiArt/EmojiArtModel.swift [hunk 4]
```diff
@@ -0,0 +1,37 @@
+//
+//  EmojiArtModel.swift
+//  EmojiArt
+//
+//  Created by Izabella Julia dos Santos on 20/06/22.
+//
+
+import Foundation
+
+struct EmojiArtModel {
```
Comment (RenatoMAssis) [inline EmojiArt/EmojiArtModel.swift:10]: same for private stuffs

File: EmojiArt/UtilityExtensions.swift [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    var center: CGPoint {
+        CGPoint(x: midX, y: midY)
+    }
+}
+
+extension Character {
+    var isEmoji: Bool {
+        if let firstScalar = unicodeScalars.first, firstScalar.properties.isEmoji {
+            return (firstScalar.value >= 0x238d || unicodeScalars.count > 1)
+        } else {
```
Comment (RenatoMAssis) [inline EmojiArt/UtilityExtensions.swift:26]: you don't need this else.......

**Pull request merged**